### ISSUE TITLE
[good first issue - beginner] Ensure all examples have a jupyter notebook to present the example

### ISSUE THREAD

**Issue by skrawcz** [DAGWorks-Inc/hamilton#247]
**Is your feature request related to a problem? Please describe.**
A lot of DS want a notebook to get started with, not a python script -- these can be too intimidating.

**Describe the solution you'd like**
We should make sure our examples all have a jupyter notebook to mirror the "run.py" file. That way people can see how things are exercised and then they can use a notebook to try things out.

E.g. we want each example to have a notebook kind of like the [hello world example](https://github.com/DAGWorks-Inc/hamilton/blob/main/examples/hello_world/my_notebook.ipynb).

**Extra instructions**
1. Claim an example.
2. Some example might not be applicable for a notebook -- if so we should just note it and move on.

**Examples to check off from the [/examples](https://github.com/DAGWorks-Inc/hamilton/tree/main/examples) folder**
- [ ] airflow
- [ ] async
- [x] caching_nodes (claimed @alti-tude)
- [x] dask (claimed @bryangalindo )
- [x] #1122 (@eoagyen )
- [ ] data_quality
- [ ] dbt
- [ ] feature_engineering_multiple_contexts 
- [x] hello_world
- [x] lineage
- [ ] LLM_Workflows (there is lots under here, so will break it out as we get people looking at them)
- [ ] model_examples
- [x] numpy (claimed @AnupJoseph )
- [x] polars (claimed @bryangalindo )
- [x] prefect (claimed @flaviassantos )
- [ ] ray 
- [x] #1179 (@eoagyen )
- [x] scikit-learn (claimed @AnupJoseph )
- [ ] spark

**Comment by AnupJoseph** [comment]
Hi! Is anyone working on this? Can I claim the numpy and scikit-learn  portions of this task?

**Comment by skrawcz** [comment]
> Hi! Is anyone working on this? Can I claim the numpy and scikit-learn portions of this task?

Sure. I'll assign your name to them.

**Comment by skrawcz** [comment]
@AnupJoseph thanks for those two! Let me know if you want to claim more, or if you want something a little "harder".

**Comment by AnupJoseph** [comment]
> @AnupJoseph thanks for those two! Let me know if you want to claim more, or if you want something a little "harder".

Welcome! I'll take up some of the other examples as I get the time

**Comment by alti-tude** [comment]
Hi @skrawcz. I would like to claim `caching_nodes` and `lineage` examples if no one is working on those.

**Comment by skrawcz** [comment]
>  I would like to claim caching_nodes and lineage examples if no one is working on those.

Yep -- actually it seems like `lineage` is already done (my bad).  But `caching_nodes` isn't. So will add you to `caching_nodes` and feel free to pick another one.

**Comment by bryangalindo** [comment]
I'm open to taking any of the below. Thanks!
- airflow
- dask
- dbt
- polars
- spark

**Comment by skrawcz** [comment]
@bryangalindo let me give you polars.

**Comment by skrawcz** [comment]
@alti-tude any updates on your end? Need any help?

**Comment by bryangalindo** [comment]
@skrawcz I'm working on the Dask example instead since the Polars example doesn't have a **run.py** module (unless I missed it or someone is hiding it in a shelf 🫣). I confirmed a **run.py** module exists in _examples/dask/hello_world_.

**Comment by skrawcz** [comment]
> @skrawcz I'm working on the Dask example instead since the Polars example doesn't have a **run.py** module (unless I missed it or someone is hiding it in a shelf 😄). I confirmed a **run.py** module exists in _examples/dask/hello_world_.

Ah yeah -- it might be called `my_script.py`. Just a naming difference. If you look at the code it should look fairly similar. No worries.

**Comment by bryangalindo** [comment]
> > @skrawcz I'm working on the Dask example instead since the Polars example doesn't have a **run.py** module (unless I missed it or someone is hiding it in a shelf 😄). I confirmed a **run.py** module exists in _examples/dask/hello_world_.
> 
> 
> 
> Ah yeah -- it might be called `my_script.py`. Just a naming difference. If you look at the code it should look fairly similar. No worries.

Ah thanks for pointing that out. I can continue working on the Polars example. Thanks @skrawcz!

**Comment by benhhack** [comment]
@skrawcz I can take the feature_engineering_multiple_contexts

**Comment by bryangalindo** [comment]
@skrawcz i can take the ray example

**Comment by flaviassantos** [comment]
@skrawcz can I take spark and prefect?

**Comment by CyberLogan** [comment]
are there still examples left, i would love to contribute in it

**Comment by flaviassantos** [comment]
@skrawcz, because of #241  it make sense for me to work on the LLM_Workflows as well :) Can you assigme that?

**Comment by flaviassantos** [comment]
@skrawcz can I take model_examples and LLM_workflows? I am going through the code anyway, I might as well fix a notebook for them  :) thanks

**Comment by eoagyen** [comment]
@skrawcz can I take the data_loaders?

**Comment by eoagyen** [comment]
@skrawcz can I take the data_loaders?

**Comment by skrawcz** [comment]
> @skrawcz can I take the data_loaders?

Sure!  Please comment on #1122  and I can assign it that way.

**Comment by eoagyen** [comment]
@skrawcz I would like to take the reusing_functions

**Comment by skrawcz** [comment]
@eoagyen please comment on #1179
